### ISSUE TITLE
--parallel-workers not working as expected

### ISSUE THREAD

**Issue by matteogrechi** [colcon/colcon-parallel-executor#36]
When running `colcon build --parallel-workers 1` on cpp code the number of compiler threads is not 1.
To have the "expected" behaviour I should run instead 
```
export MAKEFLAGS=-j1
colcon build --parallel-workers 1
```
Is this behaviour intended?
Is it possible to cite this fact directly in the colcon documentation?

Thank you very much!

**Comment by christophebedard** [comment]
As far as I know, that is the expected behaviour with colcon. `--parallel-workers` controls the number of "jobs" (i.e., packages in this case) being built in parallel, but then each package itself may or may not be built in parallel using multiple threads. That part depends on the underlying built system, e.g., CMake. As you've found, using the `MAKEFLAGS` environment variable is the proper way to control this for CMake packages.

**Comment by christophebedard** [comment]
`--executor sequential` only means that `colcon` will process (e.g., build or test) 1 package at a time.

**Comment by nyxrobotics** [comment]
I have heard that colcon is standard in ROS2, why is such an important issue still unresolved?
Do ROS2 users use other build systems?

**Comment by Ryanf55** [comment]
no, we just keep increasing our runner CPU/RAM when builds get OOM-killed. It's not sustainable though.

**Comment by ardabbour** [comment]
Did someone actually think about this and say, "Yes, this makes sense?"
